wordpress-mobile/WordPress-iOS
# Blogging Prompts: Local notification scheduler (part 1)

Refs #18542

This adds the core scheduler component to register local notifications. I've split the work into three parts to make the changes a bit more digestible. For this PR, here are the highlights:

- Some minor changes on `BloggingPromptsService`: I've bumped up the default `number` value from 24 to 25. This is to account for today and the next 14 days.
- Added `PromptRemindersScheduler`. This accepts `BloggingRemindersScheduler.Schedule` data structure, fetches the prompts, and bulk-registers local notifications.
- The unit tests should cover the various tricky time-related scenarios. Let me know if you've thought of any other cases that I've missed!
- I've added some TODO comments within the scheduler code, but to summarize, here are the TODOs:
  - Save notification identifiers to the local store. This is used to cancel pending/upcoming notifications.
  - Implement `unschedule` to delete pending local notifications.
  - Schedule static notifications after the prompts have run out.

## To Test

Since this is pure logic and not hooked anywhere yet, please run the unit tests and ensure they pass. 

## Regression Notes
1. Potential unintended areas of impact
N/A. The component is still isolated, and feature is unreleased.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A. The component is still isolated, and feature is unreleased.

3. What automated tests I added (or what prevented me from doing so)
Added unit tests for the scheduler logic.

PR submission checklist:
